### PR TITLE
Setting up documentation build system

### PR DESCRIPTION
**Pull request by briandoconnor**
**This Pull Request is ready for comments and feedback, I'd like to merge soon**

This addresses issue #194 via [feature/issue-194](/ga4gh/data-repository-service-schemas/tree/feature/issue-194) which is based on [feature/issue-192](/ga4gh/data-repository-service-schemas/tree/feature/issue-192).  Feature branch 192 covers some light changes to rename entities from DOS to DRS.  The latter is being evaluated by @natanlao via [PR 193](https://github.com/ga4gh/data-repository-service-schemas/pull/193).

Normally I think we should follow [HubFlow](https://datasift.github.io/gitflow/) where each feature branch is off of `develop` and each one is evaluated independently.  However, I'm making feature branches from feature branches (and making pull requests between them) since I'm at a hackathon and I don't have time to wait for each merge to develop to take place.  But at the same time I don't want to put all the hackathon changes in a single pull request, hence the chain.  Someone tell me if there's a better way to do this since on feature branch depends on the changes of the previous so I can keep coding the next feature.  Advice from someone with more experience here would be great!!

The goal of the code here is to automate the building of Swagger and HTML/PDF documentation for DRS using the code from @jaeddy and the WES repo.  I adapted it lightly, it's mostly unchanged but I made some documentation changes to reflect DRS.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -35,6 +35,25 @@ jobs:
         # Travis will install requirements.txt by default
         - flake8 --select=E121,E123,E126,E226,E24,E704,W503,W504 --ignore=E501 app.py tests
 
+    # the build process from James Eddy adapted from the WES repo
```
Comment (jaeddy) [inline .travis.yml]: Just a note: the steps/stages described under `jobs` should be able to replace the `python` and `stages` sections above. Might be a task for a separate PR.

File: DOCSBUILD.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+  + swagger.json — spec (JSON) preview for current version of [branch-name]
+  + swagger.yaml — spec (YAML) preview for current version of [branch-name]
+
++ When changes are pushed to branches on a fork of the main repo (and the user has set up Travis for their forked repo), the same path apply but should be relative to "[user-or-org].github.io/[repo]/".
+
++ `README.md` and `CONTRIBUTING.md` updated with all of the above links
+
++ `README.md` badges indicating Travis CI build status and Swagger/OpenAPI validation status
+
+**Note:** This set up was the result of substantial trial and error. It includes tools/solutions based in languages I don't normally use (Gradle, Node). There should be plenty of room for improvement and cleanup — so suggestions are welcome!
```
Comment (jaeddy) [inline DOCSBUILD.md]: Not sure if you want to include my editorial notes in the these docs. :)

File: README.md [hunk 3]
```diff
@@ -29,7 +54,11 @@ $ pip install ga4gh-dos-schemas
 ```
 
 This will install both a demonstration server and a Python client that will allow you to
-manage Data Objects in a local server. You can start the demo server using `ga4gh_dos_server`.
+manage Data Objects in a local server.
+
+## Sample Service
+
+You can start the demo server using `ga4gh_dos_server`.
```
Comment (jaeddy) [inline README.md]: This should be `ga4gh_drs_server`.

File: gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+var gulp = require('gulp');
```
Comment (jaeddy) [inline gulpfile.js]: I'm about 95% sure that this file isn't needed anymore. It's leftover from when I was using a separate **node** job in Travis to build the Swagger UI.

File: DOCSBUILD.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
+# Documentation Build Process
```
Comment (natanlao) [inline DOCSBUILD.md]: It might be a good idea to merge this document with the documentation we have on [ReadTheDocs](https://data-object-service.readthedocs.io/en/latest/contributors.html)

File: DOCSBUILD.md (hunk 2 above)
Comment (briandoconnor) [reply DOCSBUILD.md]: OK, removed

File: README.md (hunk 3 above)
Comment (briandoconnor) [reply README.md]: OK, I updated the name but the sample service may be slightly broken, need to talk to Natan

File: gulpfile.js (hunk 4 above)
Comment (briandoconnor) [reply gulpfile.js]: I'm not certain if you mean the whole js file is not needed or just this include.

I'm adding this as a task for https://github.com/ga4gh/data-repository-service-schemas/issues/203

**Comment by briandoconnor** [comment]
Hey @jaeddy and @natanlao, thanks so much for reviewing.  I made a few small changes and pushed a couple of James' comments to a ticket (https://github.com/ga4gh/data-repository-service-schemas/issues/203) for a future PR.  I'm going to go ahead and merge this into my other pull request.
